liana-gui/src/download.rs

```diff
@@ -1,135 +1,82 @@
 // This is based on https://github.com/iced-rs/iced/blob/master/examples/download_progress/src/download.rs
 // with some modifications to store the downloaded bytes in `Progress::Finished` and `State::Downloading`
 // and to keep track of any download errors.
-use iced::subscription;
+use iced::futures::{SinkExt, Stream, StreamExt};
+use iced::stream::try_channel;
+use iced::Subscription;
 
 use std::hash::Hash;
+use std::sync::Arc;
 
 // Just a little utility function
 pub fn file<I: 'static + Hash + Copy + Send + Sync, T: ToString>(
     id: I,
     url: T,
-) -> iced::Subscription<(I, Progress)> {
-    subscription::unfold(id, State::Ready(url.to_string()), move |state| {
-        download(id, state)
+) -> iced::Subscription<(I, Result<Progress, DownloadError>)> {
+    Subscription::run_with_id(
+        id,
+        download(url.to_string()).map(move |progress| (id, progress)),
+    )
+}
+
+fn download(url: String) -> impl Stream<Item = Result<Progress, DownloadError>> {
+    try_channel(1, move |mut output| async move {
```

Comment:
```suggestion
    try_channel(100, move |mut output| async move {
```
i'm really not confident to ave a size of 1 here, especially as we expect a bunch of data and we are not in control of the other end of the channel